src/main/java/htsjdk/samtools/cram/structure/SliceIO.java

```diff
@@ -105,17 +106,20 @@ private static void parseSliceHeaderBlock(final int major, final Slice slice) th
     }
 
     private static void readSliceBlocks(final int major, final Slice slice, final InputStream inputStream) throws IOException {
-        slice.external = new HashMap<Integer, Block>();
+        slice.external = new HashMap<>();
         for (int i = 0; i < slice.nofBlocks; i++) {
-            final Block block = Block.readFromInputStream(major, inputStream);
+            final Block block = Block.read(major, inputStream);
 
             switch (block.getContentType()) {
                 case CORE:
                     slice.coreBlock = block;
                     break;
                 case EXTERNAL:
-                    if (slice.embeddedRefBlockContentID == block.getContentId()) slice.embeddedRefBlock = block;
-                    slice.external.put(block.getContentId(), block);
+                    final ExternalDataBlock extBlock = (ExternalDataBlock) block;
```

Comment:
It does, because this `block` comes from the generic `Block.read()`.  If we made `external` and `embeddedRefBlock` into simple `Block`s we could remove this.  I prefer them as `ExternalDataBlock`s.

Reply:
changed it to `Block`